mathics/builtin/numbers/calculus.py

```diff
@@ -1957,25 +1957,28 @@ def get_accuracy_and_prec(opts: dict, evaluation: "Evaluation"):
     Looks at an opts dictionary and tries to determine the numeric values of
     Accuracy and Precision goals. If not available, returns None.
     """
-    acc_goal = opts.get("System`AccuracyGoal", None)
-    if acc_goal:
-        acc_goal = apply_N(acc_goal, evaluation)
-        if acc_goal is SymbolAutomatic:
-            acc_goal = Real(12.0)
-        elif acc_goal is SymbolInfinity:
-            acc_goal = None
-        elif not isinstance(acc_goal, Number):
-            acc_goal = None
+    # comment @mmatera: I fix the default value for Accuracy
+    # and Precision goals to 12 because it ensures that
+    # the results of the tests coincides with WMA upto
+    # 6 digits. In any case, probably the default value should be
+    # determined inside the methods that implements the specific
+    # solvers.
+
+    def to_number_or_none(value):
+        if value:
+            value = apply_N(value, evaluation)
+        if value is SymbolAutomatic:
+            value = Real(12.0)
+        elif value is SymbolInfinity:
+            value = None
+        elif not isinstance(value, Number):
+            value = None
+        return value
 
+    acc_goal = opts.get("System`AccuracyGoal", None)
+    acc_goal = to_number_or_none(acc_goal)
```

Comment:
Please don't use the same name which changes type. In a strongly typed language this would be disallowed. But even in a language that isn't strongly typed, I think it is confusing

Reply:
what about this?

```
acc_goal_opt = opts.get("System`AccuracyGoal", None)
acc_goal = to_number_or_none(acc_goal_opt)
```
still `acc_goal` would be of type `typing.Union[None,Real]`  . Would that be acceptable?

Reply:
Great - thanks for your patience and understanding. It  be would be to add that signature too on the function definition

```python
def to_number_or_node( ... ) -> Optional[Real]
```